_nx_parallel/__init__.py

```diff
@@ -13,6 +13,11 @@ def get_info():
         "short_summary": "Parallel backend for NetworkX algorithms",
         "default_config": _config,
         "functions": {
+            "_set_nx_config": {
+                "url": "https://github.com/networkx/nx-parallel/blob/main/nx_parallel/utils/decorators.py#L12",
+                "additional_docs": "Decorator to set the configuration for the parallel computation of the nx-parallel algorithms.",
+                "additional_parameters": None,
+            },
```

Comment:
hmmm..  I am surprised that this shows up here because it is a private function.
Would it be better to have `get_info` only include the public functions?  The private functions should not be backends for networkx right?

Actually, I think the `get_info` probably shouldn't include any of the `utils` functions... Maybe just the functions that are backend functions for networkx...  But maybe I am asking too much of this system... :) Let's work on this/that in a later PR.

Reply:
> hmmm.. I am surprised that this shows up here because it is a private function. Would it be better to have `get_info` only include the public functions? The private functions should not be backends for networkx right?

If by "private function" you mean that `_set_nx_config` is not in `__all__`, then that's not true, bcoz it is.

> Actually, I think the `get_info` probably shouldn't include any of the `utils` functions... Maybe just the functions that are backend functions for networkx... But maybe I am asking too much of this system... :) Let's work on this/that in a later PR.

Since none of the functions in networkx's utils folder have a `_dispatchable` decorator, I created PR https://github.com/networkx/nx-parallel/pull/78 that skips functions in `utils` while creating the `get_info` dict.

Reply:
I was saying that `_set_nx_config` is private because it starts with an underscore (which seems to be a standard way to indicate a function that is not part of the public API). Perhaps we shouldn't be putting `_set_nx_config` into the `__all__` attribute.  I guess if we always call it as `@nxp.utils._set_nx_config` it doesn't need to be in `__all__` we just import it explicitly as part of `utils.__init__.py` with something like `from decorators import _set_nx_config`.  Then, even though utils is imported (via *) in the main `__init__.py`, my understanding is that `_set_nx_config` will not be added to the main namespace. But it would be worth checking that.

Reply:
ref. https://github.com/networkx/nx-parallel/pull/75#discussion_r1728620575